File: openlineplanner_backend/helpers/geometry.py
import math

def get_distance(point_one, point_two):
  lng_1 = point_one["lng"]
  lat_1 = point_one["lat"]
  lng_2 = point_two["lng"]
  lat_2 = point_two["lat"]
  lng_1, lat_1, lng_2, lat_2 = map(math.radians, [lng_1, lat_1, lng_2, lat_2])
  d_lat = lat_2 - lat_1
  d_lng = lng_2 - lng_1 

  temp = (  
      math.sin(d_lat / 2) ** 2 
    + math.cos(lat_1) 
    * math.cos(lat_2) 
    * math.sin(d_lng / 2) ** 2
  )

  return int(6373.0 * (2 * math.atan2(math.sqrt(temp), math.sqrt(1 - temp))) * 1000)

File: openlineplanner/src/helpers/geometry.js
import gedesic from "geographiclib-geodesic";

export const calculateMidPoint = (latLngA, latLngB) => {
  const geod84 = gedesic.Geodesic.WGS84;
  const totalDistance = Math.round(geod84.Inverse(latLngA.lat, latLngA.lng, latLngB.lat, latLngB.lng).s12);
  console.log(geod84.Inverse(latLngA.lat, latLngA.lng, latLngB.lat, latLngB.lng));
  console.log(totalDistance);

  function toRadians(degrees) {
    return degrees * (Math.PI / 180);
  }

  function toDegrees(radians) {
    return (radians * (180 / Math.PI)).toFixed(4);
  }

  const lngDiff = toRadians(latLngB.lng - latLngA.lng);
  const latA = toRadians(latLngA.lat);
  const latB = toRadians(latLngB.lat);
  const lngA = toRadians(latLngA.lng);

  const bx = Math.cos(latB) * Math.cos(lngDiff);
  const by = Math.cos(latB) * Math.sin(lngDiff);

  const latMidway = toDegrees(
    Math.atan2(
      Math.sin(latA) + Math.sin(latB),
      Math.sqrt((Math.cos(latA) + bx) * (Math.cos(latA) + bx) + by * by)
    )
  );
  const lngMidway = toDegrees(lngA + Math.atan2(by, Math.cos(latA) + bx));

  return { lat: latMidway, lng: lngMidway, totalDistance};
};
